### ISSUE TITLE
Allow users to mark addresses as favorite

### ISSUE THREAD

**Issue by wmitsuda** [otterscan/otterscan#1006]
Most obvious use case: star your own addresses.

Where should we store that info? local storage seems obvious, although browse purge policies may mess with that, should test it.

**Comment by sealer3** [comment]
It sounds like we need a generic, abstract, and persistent per-user storage class which could store and retrieve favorites, address labels, and other settings. Personally I would swap the backend for one that stores the settings permanently in a database for Otterscan instances targeting only one user.

**Comment by wmitsuda** [comment]
store it in google drive? 😅

**Comment by sealer3** [comment]
No, I was just going to use something a simple server running on a different port that stores address labels and other personal settings in a database. I access an Otterscan instance from more than one device, so local storage won't work for me.

As long as the interface is abstract, it will at least be easy for me to swap out the backend implementation.

**Comment by JamesHinshelwood** [comment]
I agree its makes sense to support multiple backends for persistent storage. Though I think for the first iteration, just supporting local storage would be perfectly fine. Honestly Google Drive isn't even that bad an idea - e.g. check out the storage options at https://app.diagrams.net/
